e2e/docs/E2E_FRAMEWORK.md

```diff
@@ -0,0 +1,229 @@
+
+**DevNet** - development network that acts as a rough staging environment.
+
+**TestNet** - a production-like test network targeted for launch by the end of the summer of 2023 that is more permanent on principle.
+
+**Gherkin** - a syntax for writing behavior tests that match to function handlers allowing declarative behavioral tests to be written for a given piece of software. They follow the `Given/When/Then` format.
+
+**Cucumber** - Gherkin syntax (above) is used to write Cucumber tests.
+
+**B.D.D.** - Behavior-driven development. A type of testing that asserts on the behavior the user should experience when interacting with the program under test. It is typically used in close relationship with a product development team for this reason.
+
+**Kubernetes** - a container orchestration platform that LocalNet uses to manage the development environment for POKT.
+
+**Tilt** - allows for fine-grained container scripting in Kubernetes environments including automatic restarts and rebuilds of Docker resources.
+
+## Proposed Solution
+
+The proposed solution is to run Cucumber behavior tests against the LocalNet.
+
+1. Run a LocalNet instance
+2. Take control of the `v1-client` pod
+3. Feed that pod RPC messages via `kubectl`
+4. Evaluate the success or error and subsequent network conditions.
+
+### Root Feature Example
+
+Below is an example of testing the `help` command of the Pocket binary.
+
+```gherkin
+Feature: Root Namespace
+
+  Scenario: User Needs Help
+    Given the user has a validator
+    When the user runs the command "help"
+    Then the user should be able to see standard output containing "Available Commands"
+    And the pocket client should have exited without error
+```
+
+Scenarios are specific sets of `Given When Then` step definitions and are composable to some degree. [Step definitions](https://cucumber.io/docs/cucumber/step-definitions/) are functions with a regular expression that links it to a Gherkin step.
+
+The goal is to specify a handful of basic but flexible step definitions that are composed into more complex scenarios with only Gherkin tests being written to define functionality.
+
+Adaptability of the `RunCommand` method is the crux of this design.
+
+```go
+// the step regex is registered in the beginning of the tests.
+ctx.Step(`^the user runs the command "([^"]*)"$`, theUserRunsTheCommand)
+// and the associated Go function that it targets and the cmd arg it takes
+func theUserRunsTheCommand(cmd string) error {
+    cmds := strings.Split(cmd, " ")
+    result, err := validator.RunCommand(cmds...)
+    if err != nil {
+        validator.result = result
+        return err
+    }
+    if result.Err != nil {
+        return result.Err
+    }
+    return nil
+}
+```
+
+- The above step definition function is incredibly powerful in that it allows the Validator to compose and execute commands on the Pod. The tests form this into a step definition for staking, unstaking, & help command validation, etc…
```

Comment:
Optional. I think this will help

```suggestion
- The above step definition function is incredibly powerful in that it allows the `Validator` to compose and execute commands on the `Pod`. The tests form this into a step definition for staking, unstaking, & help command validation, etc…
```